### ISSUE TITLE
NVDA silent when renaming a file

### ISSUE THREAD

**Issue by andrew-l-d** [nvaccess/nvda#16991]
### Steps to reproduce:
1. Open Windows Explorer
2. Navigate to a file to be renamed
3. Press F2

### Actual behavior:

C:\Users\Andrew\Documents\nvda\NVDA
Items View  list
message.html  82 of 110
up arrow
libssl-3.dll  81 of 110
down arrow
message.html  82 of 110
f2
right arrow
right arrow
right arrow
right arrow
left arrow
left arrow
left arrow
left arrow
escape

### Expected behavior:

NVDA should announce that there is an edit field and pressing right and left arrows should announce characters in the filename.
Cannot demonstrate with Speech Viewer without loading an earlier version of NVDA.
### NVDA logs, crash dumps and other attachments:
Log fiel attached.  I pressed F2 and arrowed around, but nothing seems to show up in the log file.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
NVDA 2024 v3 RC6
#### Windows version:
Windows 11
#### Name and version of other software in use when reproducing the issue:
Windows Explorer
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Tried 2024 v3 Beta 3 on a computer running Windows 10 with the same result.  Surprised I had not come across the issue earlier.  Definitely works correctly with 2024 v2.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not tried that.
[logfile.log](https://github.com/user-attachments/files/16579291/logfile.log)

**Comment by XLTechie** [comment]
>     Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
> Not tried that.

Why? That is probably the solution. This is not an optional problem solving step either way.

If running it doesn't help,, please set your log level to debug in NVDA general settings, and generate another log.

**Comment by tarikhadzirovicofficial** [comment]
Have you tried pressing the down arrow after pressing F2 to see if the file name will be read?
I’ve experienced the same issue, but it happens everywhere where text can be read with the arrow keys (whether it’s read-only or editable). However, the problem would always be resolved after restarting NVDA. For example, when pressing the Home or End key, neither the first nor the last character would be read, and the same goes for Control+Home/End (the focus would shift to the first line, but the text of that line wouldn’t be read). The only way I could read the text was by using the up and down arrows. Try that, as well as what I mentioned with the Home and End keys.

**Comment by seanbudd** [comment]
Please run the COM registration fixing tool, restart device and report back.
If you can still reproduce this issue and provide a [log file](https://github.com/nvaccess/nvda/wiki/LogFilesAndCrashDumps) of the behaviour. Ensure your log level is set to debug in general preferences.

Kind Regards,
NV Access Software Developers

**Comment by tarikhadzirovicofficial** [comment]
Hi @seanbudd !
I don’t know exactly what causes the problem, and it happens very rarely. I have already run the COM registration fixing tool earlier, but even after restarting the device, the problem occurred several times. However, the next time it happens, I will attach the log file here.

**Comment by andrew-l-d** [comment]
Apologies for not running the COM registration fixing tool yesterday.  I was in a desperate hurry to prepare a presentation at the time.
Excuses dealt with, today I cannot replicate the issue.  My concern last night was that the problem occurred with both RC1 on this machine and beta 3 on another.  I have never had a similar problem with previous versions and was aware of the likely imminent release of 2024.3.
If the problem does not occur again in the next day or so I will close the ticket.  Otherwise, will report again after running the COM registration fixing tool.

Andrew

**Comment by SaschaCowley** [comment]
This (or something similar) happened for me last night. The issue for me was that for some reason Explorer's IAccessible server had disconnected from NVDA. It is worth noting that I was very rapidly pressing the right arrow, so perhaps it disconnected due to being flooded with requests. Log fragment attached.
[log.txt](https://github.com/user-attachments/files/16861527/log.txt)

**Comment by Adriani90** [comment]
@SaschaCowley did you see this issue hapenning with NVDA last 64bit alpha and Windows 11 24 H2 or 25 H2?
I cannot reproduce it.

**Comment by SaschaCowley** [comment]
@Adriani90 I can't reproduce right now, no. However, as noted, this is an intermittent issue, and we certainly do have intermittent issues where NVDA will just stop working correctly in editable text.

**Comment by seanbudd** [comment]
Can anyone still reproduce this in 2026.3 alphas?

**Comment by makhlwf** [comment]
> Can anyone still reproduce this in 2026.3 alphas?

yes, i can, running NVDA version alpha-57360,6864af60 (2026.3.0.57360)

**Comment by tareh7z** [comment]
Maybe it could be related to #17002.

**Comment by tareh7z** [comment]
Hi!

 can't reproduce it on [NVDA alpha-57376,e2c58a1b](https://download.nvaccess.org/snapshots/alpha/nvda_snapshot_alpha-57376,e2c58a1b.exe)

**Comment by makhlwf** [comment]
i tryed on another PC using the same nvda alfa version alpha-57360,6864af60 (2026.3.0.57360), and the issue does not exist cannot reproduce . how ever i see an issue when going to create a file using `ctrl+shift+n` nvda will not recognize the text input feeld and will not speek the typed words, could this be related to the same version?, i am running Windows 11 25H2 (AMD64) build 26220.8925.

edit: i ran the System Accessibility Repair Tool... on the current pc witch i faced the issue of not detecting the rename input box and it has been fixed as well, but the issue of the new file input box still exist

**Comment by andrew-l-d** [comment]
I had completely forgotten about the issue and I use NVDA on several machines.  Have not used the alpha but no problems for at least 1.5 years on any versions up to 2026.2 beta 10.  Just checked on my Snapdragon machine and no problems there.  Also no problems when using shift+ctrl+n to create a new folder.
